pkg/actions/service.go

```diff
@@ -326,117 +418,117 @@ func buildRunManifestFromTasks(record graveler.HookRecord, tasks [][]*Task) RunM
 }
 
 // UpdateCommitID assume record is a post event, we use the PreRunID to update the commit_id and save the run manifest again
-func (s *Service) UpdateCommitID(ctx context.Context, repositoryID string, storageNamespace string, runID string, commitID string) error {
+func (s *KVService) UpdateCommitID(ctx context.Context, repositoryID string, storageNamespace string, runID string, commitID string) error {
 	if runID == "" {
 		return fmt.Errorf("run id: %w", ErrNotFound)
 	}
 
-	// update database and re-read the run manifest
-	var manifest *RunManifest
-	_, err := s.DB.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		// update commit id
-		res, err := tx.Exec(`UPDATE actions_runs SET commit_id=$3 WHERE repository_id=$1 AND run_id=$2`,
-			repositoryID, runID, commitID)
-		if err != nil {
-			return nil, fmt.Errorf("update run commit_id: %w", err)
-		}
-		// return if nothing was updated
-		if res.RowsAffected() == 0 {
-			return nil, nil
+	runKey := getRunPath(repositoryID, runID)
+	run := RunResultData{}
+	_, err := s.Store.GetMsg(ctx, runKey, &run)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			return nil // nothing to update
```

Comment:
How is this possible?

Reply:
No real way for this to happen currently - I'll remove this check

Reply:
Please note that this is a change of behavior from DB service implementation when UPDATE returns no rows affected. As result I must remove some test logic in service_test so we won't fail on it